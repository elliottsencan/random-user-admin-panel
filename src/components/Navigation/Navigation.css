/*should get sass wired up here */

aside.navigation nav ul {
	list-style: none;
	display: flex;
	flex-direction: column;
	width: 100%;
	opacity: 1;
}

aside.navigation nav ul li{
	flex: 1 1 40px;
}

aside.navigation nav ul li a {
	text-decoration: none;
}