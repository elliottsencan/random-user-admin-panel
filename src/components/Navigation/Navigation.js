import React from 'react';
import { NavLink } from 'react-router-dom';
import './Navigation.css';
import ActionTrendingUp from 'material-ui/svg-icons/action/trending-up';
import ActionPermIdentity from 'material-ui/svg-icons/action/perm-identity';
import ActionAssessment from 'material-ui/svg-icons/action/assessment';

/**
 * Contains application routes
 */
const Navigation = () => ( <aside className='navigation'>
  <nav>
    <ul>
      <li>
        <NavLink to="/" exact={true}><ActionTrendingUp className="nav-icon"/>My dashboard</NavLink>
      </li>
      <li>
        <NavLink to="/accounts"><ActionPermIdentity className="nav-icon"/>Accounts</NavLink>
      </li>
      <li>
        <NavLink to="/bills"><ActionAssessment className="nav-icon"/>Bills</NavLink>
      </li>
    </ul>
  </nav>
</aside> );

Navigation.displayName = 'Navigation';

export default Navigation;
