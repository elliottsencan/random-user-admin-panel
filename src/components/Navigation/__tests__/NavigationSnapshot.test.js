import React from 'react';
import Navigation from '../Navigation';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure( { adapter: new Adapter() } );

it( 'renders correctly', () => {
  const tree = toJson( shallow( <Navigation/> ) );
  expect( tree ).toMatchSnapshot();
} );