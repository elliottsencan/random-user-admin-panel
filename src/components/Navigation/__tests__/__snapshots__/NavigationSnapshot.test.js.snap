// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<aside
  className="navigation"
>
  <nav>
    <ul>
      <li>
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          exact={true}
          to="/"
        >
          <ActionTrendingUp
            className="nav-icon"
          />
          My dashboard
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/accounts"
        >
          <ActionPermIdentity
            className="nav-icon"
          />
          Accounts
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/bills"
        >
          <ActionAssessment
            className="nav-icon"
          />
          Bills
        </NavLink>
      </li>
    </ul>
  </nav>
</aside>
`;
